src/JointPositionGoalController.cpp

```diff
@@ -0,0 +1,349 @@
+
+            // Check that new position is within joint position limits, enforce by clamping
+            joint_limits_interface::JointLimits &limits = joint_limits_[joint_name];
+            double q_command_with_position_limits_enforced = clamp(q_command, limits.min_position,
+                                                                   limits.max_position);
+
+//            std::cout << joint_name << std::endl;
+//            std::cout << "q: " << q << std::endl;
+//            std::cout << "q_desired: " << q_desired << std::endl;
+//            std::cout << "q_start: " << q_start << std::endl;
+//            std::cout << "q_command: " << q_command << std::endl;
+//            std::cout << "q_command_with_position_limits_enforced: " << q_command_with_position_limits_enforced <<
+//            std::endl;
+//            std::cout << std::endl;
+
+            // Write command to joint
+            iter->second.setCommand(q_command_with_position_limits_enforced);
+
+            lcm_pose_msg.joint_name[positionJointIndex] = joint_name;
+            lcm_pose_msg.joint_position[positionJointIndex] = static_cast<float>(q);
+            lcm_pose_msg.joint_velocity[positionJointIndex] = static_cast<float>(qd);
+
+            lcm_state_msg.joint_name[positionJointIndex] = joint_name;
+            lcm_state_msg.joint_position[positionJointIndex] = static_cast<float>(q);
+            lcm_state_msg.joint_velocity[positionJointIndex] = static_cast<float>(qd);
+
+            // republish to guarantee sync
+            lcm_commanded_msg.joint_name[positionJointIndex] = joint_name;
+            lcm_commanded_msg.joint_position[positionJointIndex] = static_cast<float>(q_command_with_position_limits_enforced);
+
+            positionJointIndex++;
+        }
+
+        lcm_->publish("VAL_CORE_ROBOT_STATE", &lcm_pose_msg);
+        lcm_->publish("VAL_COMMAND_FEEDBACK", &lcm_commanded_msg);
+
+        if (publish_est_robot_state_)
+            lcm_->publish("EST_ROBOT_STATE", &lcm_state_msg);
+    }
+
+    void JointPositionGoalController::stopping(const ros::Time &time) { }
+
+    JointPositionGoalController_LCMHandler::JointPositionGoalController_LCMHandler(JointPositionGoalController &parent)
+            : parent_(parent) {
+        lcm_ = std::shared_ptr<lcm::LCM>(new lcm::LCM);
+        if (!lcm_->good()) {
+            std::cerr << "ERROR: handler lcm is not good()" << std::endl;
+        }
+        lcm_->subscribe("JOINT_POSITION_GOAL", &JointPositionGoalController_LCMHandler::jointPositionGoalHandler, this);
+    }
+
+    JointPositionGoalController_LCMHandler::~JointPositionGoalController_LCMHandler() { }
+
+    void JointPositionGoalController_LCMHandler::jointPositionGoalHandler(const lcm::ReceiveBuffer *rbuf,
+                                                                          const std::string &channel,
+                                                                          const bot_core::robot_state_t *msg) {
```

Comment:
i don't think this is the message we want to use in this API, but lets change it later